index.bs

```diff
@@ -78,4 +260,327 @@ interface StorageBucket {
 };
 </xmp>
 
+A {{StorageBucket}} has an associated [=/storage bucket=].
+
+<h3 id="storage-bucket-persistence">Persistence</h3>
+
+Issue: Merge with [[Storage#buckets]] which already defines [=bucket mode=].
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">persist()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with {{InvalidStateError}}.
+    
+    1. If |bucket|'s [=bucket mode=] is `"persistent"`, [=/resolve=] |p| with true.
+
+    1. Otherwise,
+
+        1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+        1. Let |permission| be the result of [=getting the current permission state=] with `"persistent-storage"` and |environment|.
+
+        1. If |permission| is "{{PermissionState/granted}}", then set |bucket|'s [=bucket mode=] to `"persistent"` and [=/resolve=] |p| with true.
+
+        1. Otherwise, [=/resolve=] |p| with false.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">persisted()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise,
+
+        1. If |bucket|'s [=bucket mode=] is `"persistent"`, then [=/resolve=] |p| with true.
```

Comment:
Why does this need to be in parallel?  It seems like it could all be synchronous like durability()?

Reply:
Done